src/constructor.jl

```diff
@@ -61,16 +74,27 @@ function construct_object(constructor::Constructor, node::Node)
     if haskey(constructor.yaml_constructors, node.tag)
         node_constructor = constructor.yaml_constructors[node.tag]
     else
-        # TODO: Multi-constructors. Constructors that operate on prefixes.
-
-        if haskey(constructor.yaml_constructors, nothing)
-            node_constructor = constructor.yaml_constructors[nothing]
-        elseif typeof(node) == ScalarNode
-            node_constructor = construct_scalar
-        elseif typeof(node) == SequenceNode
-            node_constructor = construct_sequence
-        elseif typeof(node) == MappingNode
-            node_constructor = construct_mapping
+        for (tag_prefix, node_const) in constructor.yaml_multi_constructors
+            if tag_prefix !== nothing && startswith(node.tag, tag_prefix)
+                tag_suffix = node.tag[length(tag_prefix) + 1:end]
+                node_constructor = node_const
+                break
+            end
+        end
+
+        if node_constructor === nothing
+            if haskey(constructor.yaml_multi_constructors, nothing)
+                tag_suffix = node.tag
+                node_constructor = constructor.yaml_multi_constructors[nothing]
+            elseif haskey(constructor.yaml_constructors, nothing)
+                node_constructor = constructor.yaml_constructors[nothing]
+            elseif typeof(node) == ScalarNode
```

Comment:
I know this is just pasted from the existing code, but let's take this opportunity to change to using `isa` instead of `typeof` and `<:` 🙂

Reply:
Done.